modules/@apostrophecms/area/ui/apos/components/AposAreaWidget.vue

```diff
@@ -395,11 +401,16 @@ export default {
       if (this.focused) {
         return;
       }
-      this.state.add.top.show = true;
-      this.state.add.bottom.show = true;
+      if (!this.foreign) {
+        this.state.add.top.show = true;
+        this.state.add.bottom.show = true;
+      }
       this.state.container.highlight = true;
       this.state.labels.show = true;
-      apos.bus.$emit('widget-hover', this.widget._id);
+      apos.bus.$emit('widget-hover', {
+        _id: this.widget._id,
+        nonForeignId: this.foreign ? this.$el.closest('[data-apos-widget-foreign="0"]').getAttribute('data-apos-widget-id') : null
```

Comment:
This is a necessary compromise to discover the closest non-foreign widget.

I tried having the parent find the child, but this doesn't work well because of chicken and egg problems. Looking back up the tree is a sound strategy.